pkg/adapter/quota.go

```diff
@@ -34,11 +34,26 @@ type (
 		Builder
 
 		// NewQuota returns a new instance of the Quota aspect.
-		NewQuota(env Env, c AspectConfig) (QuotaAspect, error)
+		NewQuota(env Env, c AspectConfig, d map[string]*QuotaDefinition) (QuotaAspect, error)
+	}
+
+	// QuotaKind determines the usage semantics
+	QuotaKind int8
+
+	// QuotaDefinition is used to describe an individual quota the aspect will encounter at runtime
+	QuotaDefinition struct {
+		// MaxAmount specifies the upper limit for the quota
+		MaxAmount int64
+
+		// Size of rolling window in seconds
+		ExpirationSeconds int32
```

Comment:
any reason not to take a generic time.Duration here?  Do we care about flexibility for sub-second windows?